Deploy more smart pointers in ExtensionStyleSheets.cpp, FragmentDirectiveRangeFinder.cpp,

rniwa opened the pull request:
#### e2b769dba64ecedffc809cfd1bf07f08d7abe9a4
<pre>
Deploy more smart pointers in ExtensionStyleSheets.cpp, FragmentDirectiveRangeFinder.cpp,
FullscreenManager.cpp, IdleDeadline.cpp, and ImageOverlay.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=264797">https://bugs.webkit.org/show_bug.cgi?id=264797</a>

Reviewed by Chris Dumez.

Deployed more smart pointers as warned by the clang static analyzer.

* Source/WebCore/dom/ExtensionStyleSheets.cpp:
(WebCore::ExtensionStyleSheets::pageUserSheet):
(WebCore::ExtensionStyleSheets::clearPageUserSheet):
(WebCore::ExtensionStyleSheets::updatePageUserSheet):
(WebCore::ExtensionStyleSheets::updateInjectedStyleSheetCache const):
(WebCore::ExtensionStyleSheets::invalidateInjectedStyleSheetCache):
(WebCore::ExtensionStyleSheets::addUserStyleSheet):
(WebCore::ExtensionStyleSheets::addAuthorStyleSheetForTesting):
(WebCore::ExtensionStyleSheets::addDisplayNoneSelector):
(WebCore::ExtensionStyleSheets::maybeAddContentExtensionSheet):
* Source/WebCore/dom/ExtensionStyleSheets.h:
* Source/WebCore/dom/FragmentDirectiveRangeFinder.cpp:
(WebCore::FragmentDirectiveRangeFinder::rangeOfStringInRange):
(WebCore::FragmentDirectiveRangeFinder::advanceRangeStartToNextNonWhitespace):
* Source/WebCore/dom/FullscreenManager.cpp:
(WebCore::FullscreenManager::protectedTopDocument):
(WebCore::FullscreenManager::finishExitFullscreen):
(WebCore::FullscreenManager::didExitFullscreen):
* Source/WebCore/dom/FullscreenManager.h:
* Source/WebCore/dom/IdleDeadline.cpp:
(WebCore::IdleDeadline::timeRemaining const):
* Source/WebCore/dom/ImageOverlay.cpp:
(WebCore::ImageOverlay::installImageOverlayStyleSheet):
(WebCore::ImageOverlay::updateSubtree):

Canonical link: <a href="https://commits.webkit.org/270750@main">https://commits.webkit.org/270750@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/50445968bcb6a7638e4eb6f9eab9c3708a243670

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26250 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4859 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27515 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28346 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24031 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/26580 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6642 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2279 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24040 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26506 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3710 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22583 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28924 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3307 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23548 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29601 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23931 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23945 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27489 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3344 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1552 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4761 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3817 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3388 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3669 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/dom/ExtensionStyleSheets.cpp:
```diff
@@ -73,15 +73,15 @@ CSSStyleSheet* ExtensionStyleSheets::pageUserSheet()
         return 0;
     
-    m_pageUserSheet = createExtensionsStyleSheet(m_document, m_document.settings().userStyleSheetLocation(), userSheetText, UserStyleLevel::User);
+    m_pageUserSheet = createExtensionsStyleSheet(m_document.get(), m_document->settings().userStyleSheetLocation(), userSheetText, UserStyleLevel::User);
```
Seems we should pass a protected document here.

rniwa replied:
Fixed.

cdumez commented on Source/WebCore/dom/ExtensionStyleSheets.cpp:
```diff
@@ -90,15 +90,15 @@ void ExtensionStyleSheets::clearPageUserSheet()
     if (m_pageUserSheet) {
         m_pageUserSheet = nullptr;
-        m_document.styleScope().didChangeStyleSheetEnvironment();
+        m_document->styleScope().didChangeStyleSheetEnvironment();
```
How come we're not protecting the document here? Also, it'd be good to call the existing `checkedStyleScope()`.

rniwa replied:
It's because `styleScope()` is a trivial function but you're right that we should be using `checkedStyleScope()` here.

cdumez replied:
styleScope() is a trivial function but we're calling didChangeStyleSheetEnvironment() on it, which is not trivial. This is why I would protect the document, in order to protect the style scope (which is not ref counted), and then get a checkedStyleScope() from the document.

rniwa replied:
Hm... okay. I guess we could do that.

cdumez commented on Source/WebCore/dom/ExtensionStyleSheets.cpp:
```diff
@@ -90,15 +90,15 @@ void ExtensionStyleSheets::clearPageUserSheet()
     clearPageUserSheet();
     if (pageUserSheet())
-        m_document.styleScope().didChangeStyleSheetEnvironment();
+        m_document->styleScope().didChangeStyleSheetEnvironment();
```
ditto.

rniwa replied:
Fixed.

cdumez commented on Source/WebCore/dom/ExtensionStyleSheets.cpp:
```diff
@@ -123,12 +123,12 @@ void ExtensionStyleSheets::updateInjectedStyleSheetCache() const
 
     auto addStyleSheet = [&](const UserStyleSheet& userStyleSheet) {
-        auto sheet = createExtensionsStyleSheet(const_cast<Document&>(m_document), userStyleSheet.url(), userStyleSheet.source(), userStyleSheet.level());
+        auto sheet = createExtensionsStyleSheet(const_cast<Document&>(m_document.get()), userStyleSheet.url(), userStyleSheet.source(), userStyleSheet.level());
```
protected document?

rniwa replied:
Fixed.

cdumez commented on Source/WebCore/dom/ExtensionStyleSheets.cpp:
```diff
@@ -123,12 +123,12 @@ void ExtensionStyleSheets::updateInjectedStyleSheetCache() const
 
     auto addStyleSheet = [&](const UserStyleSheet& userStyleSheet) {
-        auto sheet = createExtensionsStyleSheet(const_cast<Document&>(m_document), userStyleSheet.url(), userStyleSheet.source(), userStyleSheet.level());
+        auto sheet = createExtensionsStyleSheet(const_cast<Document&>(m_document.get()), userStyleSheet.url(), userStyleSheet.source(), userStyleSheet.level());
```
Could we use `Ref` instead of `auto` for clarity?

rniwa replied:
Sure.

cdumez commented on Source/WebCore/dom/ExtensionStyleSheets.cpp:
```diff
@@ -174,21 +174,21 @@ void ExtensionStyleSheets::removePageSpecificUserStyleSheet(const UserStyleSheet
 {
     m_injectedStyleSheetCacheValid = false;
-    m_document.styleScope().didChangeStyleSheetEnvironment();
+    m_document->styleScope().didChangeStyleSheetEnvironment();
```
protectedDocument() + checkedStyleScope()?

rniwa replied:
We don't need to use protectedDocument() since styleScope() is a trivial function but we should be using checkedStyleScope.

cdumez commented on Source/WebCore/dom/ExtensionStyleSheets.cpp:
```diff
@@ -174,21 +174,21 @@ void ExtensionStyleSheets::removePageSpecificUserStyleSheet(const UserStyleSheet
     ASSERT(userSheet.get().isUserStyleSheet());
     m_userStyleSheets.append(CSSStyleSheet::create(WTFMove(userSheet), m_document));
-    m_document.styleScope().didChangeStyleSheetEnvironment();
+    m_document->styleScope().didChangeStyleSheetEnvironment();
```
protectedDocument() + checkedStyleScope()?

rniwa replied:
Ditto.

cdumez commented on Source/WebCore/dom/ExtensionStyleSheets.cpp:
```diff
@@ -174,21 +174,21 @@ void ExtensionStyleSheets::removePageSpecificUserStyleSheet(const UserStyleSheet
 void ExtensionStyleSheets::addUserStyleSheet(Ref<StyleSheetContents>&& userSheet)
 {
     ASSERT(userSheet.get().isUserStyleSheet());
     m_userStyleSheets.append(CSSStyleSheet::create(WTFMove(userSheet), m_document));
```
We don't protect the document we pass here?

rniwa replied:
Fixed.

cdumez commented on Source/WebCore/dom/ExtensionStyleSheets.cpp:
```diff
@@ -174,21 +174,21 @@ void ExtensionStyleSheets::removePageSpecificUserStyleSheet(const UserStyleSheet
 void ExtensionStyleSheets::addAuthorStyleSheetForTesting(Ref<StyleSheetContents>&& authorSheet)
 {
     ASSERT(!authorSheet.get().isUserStyleSheet());
     m_authorStyleSheetsForTesting.append(CSSStyleSheet::create(WTFMove(authorSheet), m_document));
```
We don't protect the document we pass here?

rniwa replied:
Fixed.

cdumez commented on Source/WebCore/dom/ExtensionStyleSheets.cpp:
```diff
@@ -174,21 +174,21 @@ void ExtensionStyleSheets::removePageSpecificUserStyleSheet(const UserStyleSheet
     ASSERT(!authorSheet.get().isUserStyleSheet());
     m_authorStyleSheetsForTesting.append(CSSStyleSheet::create(WTFMove(authorSheet), m_document));
-    m_document.styleScope().didChangeStyleSheetEnvironment();
+    m_document->styleScope().didChangeStyleSheetEnvironment();
```
protectedDocument() + checkedStyleScope()?

rniwa replied:
Fixed.

cdumez commented on Source/WebCore/dom/ExtensionStyleSheets.cpp:
```diff
@@ -201,7 +201,7 @@ void ExtensionStyleSheets::addDisplayNoneSelector(const String& identifier, cons
 
     if (result.iterator->value->addDisplayNoneSelector(selector, selectorID))
-        m_document.styleScope().didChangeStyleSheetEnvironment();
+        m_document->styleScope().didChangeStyleSheetEnvironment();
```
protectedDocument() + checkedStyleScope()?

rniwa replied:
Fixed it to use checkedStyleScope.

cdumez commented on Source/WebCore/dom/ExtensionStyleSheets.cpp:
```diff
@@ -214,7 +214,7 @@ void ExtensionStyleSheets::maybeAddContentExtensionSheet(const String& identifie
     m_contentExtensionSheets.set(identifier, &cssSheet.get());
     m_userStyleSheets.append(adoptRef(cssSheet.leakRef()));
-    m_document.styleScope().didChangeStyleSheetEnvironment();
+    m_document->styleScope().didChangeStyleSheetEnvironment();
```
protectedDocument() + checkedStyleScope()?

rniwa replied:
Ditto.

cdumez commented on Source/WebCore/dom/ImageOverlay.cpp:
```diff
@@ -406,8 +406,9 @@ static Elements updateSubtree(HTMLElement& element, const TextRecognitionResult&
             if (line.hasTrailingNewline) {
-                lineElements.lineBreak = HTMLBRElement::create(document.get());
-                lineContainer->appendChild(*lineElements.lineBreak);
+                auto lineBreak = HTMLBRElement::create(document.get());
```
`auto` -> `Ref`?

rniwa replied:
Changed.

rniwa committed:
Deploy more smart pointers in ExtensionStyleSheets.cpp, FragmentDirectiveRangeFinder.cpp,
FullscreenManager.cpp, IdleDeadline.cpp, and ImageOverlay.cpp
https://bugs.webkit.org/show_bug.cgi?id=264797

Reviewed by Chris Dumez.

Deployed more smart pointers as warned by the clang static analyzer.

* Source/WebCore/dom/ExtensionStyleSheets.cpp:
(WebCore::ExtensionStyleSheets::pageUserSheet):
(WebCore::ExtensionStyleSheets::clearPageUserSheet):
(WebCore::ExtensionStyleSheets::updatePageUserSheet):
(WebCore::ExtensionStyleSheets::updateInjectedStyleSheetCache const):
(WebCore::ExtensionStyleSheets::invalidateInjectedStyleSheetCache):
(WebCore::ExtensionStyleSheets::addUserStyleSheet):
(WebCore::ExtensionStyleSheets::addAuthorStyleSheetForTesting):
(WebCore::ExtensionStyleSheets::addDisplayNoneSelector):
(WebCore::ExtensionStyleSheets::maybeAddContentExtensionSheet):
* Source/WebCore/dom/ExtensionStyleSheets.h:
* Source/WebCore/dom/FragmentDirectiveRangeFinder.cpp:
(WebCore::FragmentDirectiveRangeFinder::rangeOfStringInRange):
(WebCore::FragmentDirectiveRangeFinder::advanceRangeStartToNextNonWhitespace):
* Source/WebCore/dom/FullscreenManager.cpp:
(WebCore::FullscreenManager::protectedTopDocument):
(WebCore::FullscreenManager::finishExitFullscreen):
(WebCore::FullscreenManager::didExitFullscreen):
* Source/WebCore/dom/FullscreenManager.h:
* Source/WebCore/dom/IdleDeadline.cpp:
(WebCore::IdleDeadline::timeRemaining const):
* Source/WebCore/dom/ImageOverlay.cpp:
(WebCore::ImageOverlay::installImageOverlayStyleSheet):
(WebCore::ImageOverlay::updateSubtree):

Canonical link: https://commits.webkit.org/270750@main
